hls4ml/backends/vivado/vivado_backend.py

```diff
@@ -23,12 +23,39 @@ def __init__(self):
         self._register_flows()
 
     def _register_layer_attributes(self):
-        extended_attrs = {
-            SimpleRNN: [Attribute('recurrent_reuse_factor', default=1), Attribute('static', value_type=bool, default=True)],
-            LSTM: [Attribute('recurrent_reuse_factor', default=1), Attribute('static', value_type=bool, default=True)],
-            GRU: [Attribute('recurrent_reuse_factor', default=1), Attribute('static', value_type=bool, default=True)],
-        }
-        self.attribute_map.update(extended_attrs)
+        # Add RNN-specific attributes, recurrent_reuse_factor and static implementation
+        rnn_layers = [
+            SimpleRNN, LSTM, GRU,
+        ]
+
+        for layer in rnn_layers:
+            attrs = self.attribute_map.get(layer, [])
+            attrs.append(ConfigurableAttribute('recurrent_reuse_factor', default=1))
+            attrs.append(ConfigurableAttribute('static', value_type=bool, default=True))
+            self.attribute_map[layer] = attrs
+        
+        # Add ParallelizationFactor to Conv1D/2D
+        pf_layers = [
+            Conv1D, Conv2D,
+        ]
+
+        for layer in pf_layers:
+            attrs = self.attribute_map.get(layer, [])
+            attrs.append(ConfigurableAttribute('parallelization_factor', default=1))
+            self.attribute_map[layer] = attrs
+        
+        # Add ConvImplementation to Convolution+Pooling layers
+        cnn_layers = [
+            Conv1D, Conv2D,
+            SeparableConv1D, SeparableConv2D, DepthwiseConv2D,
+            Pooling1D, Pooling2D
+        ]
+
+        for layer in cnn_layers:
+            attrs = self.attribute_map.get(layer, [])
+            attrs.append(ConfigurableAttribute('conv_implementation', value_type=str, default='LineBuffer'))
```

Comment:
Should `conv_implementation` be `ConfigurableAttribute `or `ChoiceAttribute`?

Reply:
You're right, fixed.